### PR TITLE
Cable net app/empirical spring

### PR DESCRIPTION
**Pull request by KlausBSautter**
Added a very small new element for CableNetApplication, with 5 tests

### CODE REVIEW

File: applications/CableNetApplication/tests/test_empirical_spring.py [hunk 1]
```diff
@@ -0,0 +1,308 @@
+from __future__ import print_function, absolute_import, division
+import KratosMultiphysics
+
+
+import KratosMultiphysics.CableNetApplication as CableNetApplication
+import KratosMultiphysics.StructuralMechanicsApplication as StructuralMechanicsApplication
+import KratosMultiphysics.KratosUnittest as KratosUnittest
+
+from KratosMultiphysics.CableNetApplication import empirical_spring_element_process
```
Comment (philbucher) [inline applications/CableNetApplication/tests/test_empirical_spring.py:9]: this is a test, hence you have to check if numpy polywhatever is available before importing it

**Comment by KlausBSautter** [comment]
@philbucher  like that?

File: applications/CableNetApplication/custom_processes/empirical_spring_element_process.h [hunk 2]
```diff
@@ -0,0 +1,162 @@
...
+        KRATOS_CATCH("");
+    }
+
+    /// this function will be executed at every time step AFTER performing the solve phase
+    void ExecuteFinalizeSolutionStep() override
+    {
+        KRATOS_TRY;
+
+        KRATOS_CATCH("");
+    }
```
Comment (philbucher) [inline applications/CableNetApplication/custom_processes/empirical_spring_element_process.h]: I would remove the empty functions

File: applications/CableNetApplication/custom_processes/empirical_spring_element_process.h [hunk 3]
```diff
@@ -0,0 +1,162 @@
...
+
+    void CreateEmpiricalSpringElement() const
+    {
+        KRATOS_TRY;
+        Vector polynomial_order = ZeroVector(mrFittedPoly.size());
+        for (SizeType i=0;i<mrFittedPoly.size();++i) polynomial_order[i] = mrFittedPoly[i];
+        const int number_nodes = 2;
+
+        // get new element id
+        const ModelPart::IndexType new_element_id = mParameters["element_id"].GetInt();
```
Comment (philbucher) [inline applications/CableNetApplication/custom_processes/empirical_spring_element_process.h]: seems overkill
```suggestion
        const std::size_t new_element_id = mParameters["element_id"].GetInt();
```

File: applications/CableNetApplication/custom_elements/empirical_spring.cpp [hunk 4]
```diff
@@ -0,0 +1,575 @@
...
+        rElementalDofList[index + 2] =
+            GetGeometry()[i].pGetDof(DISPLACEMENT_Z);
+    }
+}
+
+void EmpiricalSpringElement3D2N::Initialize()
+{
+    KRATOS_TRY
+    KRATOS_CATCH("")
+}
```
Comment (philbucher) [inline applications/CableNetApplication/custom_elements/empirical_spring.cpp]: better remove if empty

File: applications/CableNetApplication/custom_elements/empirical_spring.cpp [hunk 5]
```diff
@@ -0,0 +1,575 @@
...
+
+int EmpiricalSpringElement3D2N::Check(const ProcessInfo& rCurrentProcessInfo)
+{
+    KRATOS_TRY
+    const double numerical_limit = std::numeric_limits<double>::epsilon();
+    const SizeType number_of_nodes = GetGeometry().size();
+    const SizeType dimension = GetGeometry().WorkingSpaceDimension();
+
+    if (dimension != msDimension ||number_of_nodes != msNumberOfNodes) {
+        KRATOS_ERROR << "The element works only in 3D and with 3 nodes" << std::endl;
```
Comment (philbucher) [inline applications/CableNetApplication/custom_elements/empirical_spring.cpp]: 3 Nodes?
Then you registered it wrong

**Review by philbucher — COMMENTED** [review]
as practice I would separate the processes into header/source

File: applications/CableNetApplication/custom_elements/empirical_spring.cpp (hunk 5 above)
Comment (KlausBSautter) [reply applications/CableNetApplication/custom_elements/empirical_spring.cpp]: the text is wrong ......

**Comment by KlausBSautter** [comment]
> as practice I would separate the processes into header/source

it only contains 1 function

**Review by philbucher — APPROVED** [review]

**Pull request merged**